app/code/community/Ebizmarts/MailChimp/Helper/Data.php

```diff
@@ -1378,40 +1378,52 @@ protected function getLastPromoCodeSent($scopeId, $scope)
      */
     public function createMergeFields($scopeId, $scope)
     {
+        $success = 1;
         $listId = $this->getGeneralList($scopeId, $scope);
         $maps = $this->unserialize($this->getMapFields($scopeId, $scope));
         $customFieldTypes = $this->unserialize($this->getCustomMergeFieldsSerialized($scopeId, $scope));
 
-        try {
-            $api = $this->getApi($scopeId, $scope);
-            $mailchimpFields = array();
-
+        if (count($maps) > 30 ) {
+            $success = 2;
+        } else {
             try {
-                $mailchimpFields = $api->getLists()->getMergeFields()->getAll($listId, null, null, 50);
-            } catch (MailChimp_Error $e) {
-                $this->logError($e->getFriendlyMessage());
-            }
+                $api = $this->getApi($scopeId, $scope);
+                $mailchimpFields = array();
 
-            if (!empty($mailchimpFields)) {
-                foreach ($maps as $map) {
-                    $customAtt = $map['magento'];
-                    $chimpTag = $map['mailchimp'];
-                    $alreadyExists = false;
+                try {
+                    $mailchimpFields = $api->getLists()->getMergeFields()->getAll(
+                        $listId,
+                        null,
+                        null,
+                        50
+                    );
+                } catch (MailChimp_Error $e) {
+                    $this->logError($e->getFriendlyMessage());
```

Comment:
When this catch is triggered, looks like the result will still be 1 which means success.
Looks like the entire logic of the results should be updated as well.
Always returning the success value from this function should be enough.